portfolio/src/main/webapp/loginCheck.js

```diff
@@ -0,0 +1,25 @@
+document.addEventListener("DOMContentLoaded", displayPage, false);
+
+function displayPage(e){
+  fetch('/loginStatus').then(response => response.json()).then(data =>{
+    if(data.loginStatus == false){
+      const body = document.getElementById("content");
+      body.innerHTML = "";
+
+      const error = document.createElement("h1");
+      error.appendChild(document.createTextNode("Error: Not logged in!"));
+      body.appendChild(error);
+
+      const loginLink = document.createElement("a");
+      loginLink.href = data.url;
+      loginLink.appendChild(document.createTextNode("Click here to log in."));
+      body.appendChild(loginLink);
```

Comment:
Not that you need to do this, but just a neat little trick. you can create these elements in the HTML and just leave them in there, but set them to be hidden. Then, if the user is not logged in, change their visibility with javascript.

Reply:
I might try this for the next PR!